pkg/typesystem/typesystem_create_test.go

```diff
@@ -0,0 +1,352 @@
+package typesystem
+
+import (
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/pkg/testutils"
+)
+
+func usersetsEquals(t *testing.T, a, b *openfgav1.Usersets) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+	childA := a.GetChild()
+	childB := b.GetChild()
+	require.Equal(t, len(childA), len(childB))
+	for i := 0; i < len(childA); i++ {
+		rewriteEquals(t, childA[i], childB[i])
+	}
+}
+
+func tupleToUsersetEquals(t *testing.T, a, b *openfgav1.TupleToUserset) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+	cuA := a.GetComputedUserset()
+	cuB := b.GetComputedUserset()
+
+	if cuA == nil || cuB == nil {
+		require.Equal(t, cuA, cuB)
+		return
+	}
+	objectRelationEquals(t, cuA, cuB)
+}
+
+func objectRelationEquals(t *testing.T, a, b *openfgav1.ObjectRelation) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+	require.Equal(t, a.GetRelation(), b.GetRelation())
+	require.Equal(t, a.GetObject(), b.GetObject())
+}
+
+func rewriteEquals(t *testing.T, a, b *openfgav1.Userset) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+
+	usersetA := a.GetUserset()
+	usersetB := b.GetUserset()
+
+	_, okA := usersetA.(*openfgav1.Userset_This)
+	_, okB := usersetB.(*openfgav1.Userset_This)
+	require.Equal(t, okA, okB, "one is Userset_This")
+
+	cuA, okA := usersetA.(*openfgav1.Userset_ComputedUserset)
+	cuB, okB := usersetB.(*openfgav1.Userset_ComputedUserset)
+	require.Equal(t, okA, okB, "one is Userset_ComputedUserset")
+	if cuA == nil || cuB == nil {
+		require.Equal(t, cuA, cuB)
+		return
+	}
+	objectRelationEquals(t, cuA.ComputedUserset, cuB.ComputedUserset)
+
+	ttuA, okA := usersetA.(*openfgav1.Userset_TupleToUserset)
+	ttuB, okB := usersetB.(*openfgav1.Userset_TupleToUserset)
+	require.Equal(t, okA, okB, "one is Userset_TupleToUserset")
+	if ttuA == nil || ttuB == nil {
+		require.Equal(t, ttuA, ttuB)
+		return
```

Comment:
Same as above - you should not return here

Reply:
fixed, thank you.